### PR TITLE
Webapp fixes by Mainul

### PR DESCRIPTION
**Pull request by yaad96**
Fixed the some following bugs - 
1. In some cases, clicking Fix on one snippet was actually sending a different snippet to the gpt
2. If a valid API key is not submitted, Fix button should not be displayed
3. If a valid API key is not submitted, the “enter API key” field is displayed, if valid key is submitted, in the top bar it is showed that “Valid API Key Submitted”
4. “Enter API Key” is displayed regardless of whether a valid api key is in local storage
5. A loading state is shown after clicking “Fix” and till the suggested snippet gets loaded

### CODE REVIEW

File: src/core/utilities.js [hunk 1]
```diff
@@ -94,10 +94,14 @@ class Utilities {
                 default:
                     break;
             }
-
+            console.log("Sending Message: ");
+            console.log(messageJson);
```
Comment (SaharMehrpour) [inline src/core/utilities.js:98]: Change it to ` console.log("Sending Message: ", messageJson);`

Comment (yaad96) [reply src/core/utilities.js]: done, please chek

File: src/core/utilities.js [hunk 2]
```diff
@@ -94,10 +94,14 @@ class Utilities {
...
             ws.send(JSON.stringify(messageJson));
-            if (secondMessageJson)
+            if (secondMessageJson){
+                console.log("Sending Second Message: ");
+                console.log(secondMessageJson);
```
Comment (SaharMehrpour) [inline src/core/utilities.js:101]: Change it to  console.log("Sending Second Message: ", secondMessageJson);

File: src/core/webSocketManager.js [hunk 3]
```diff
@@ -44,7 +44,8 @@ class WebSocketManager extends Component {
 
             let message = Utilities.parseJson(e.data, "the received message", {command: ""});
 
-            // if (message.command !== "XML") console.log(message);
+            console.log("Received message: ");
+            if (message.command !== "XML") console.log(message);
```
Comment (SaharMehrpour) [inline src/core/webSocketManager.js:48]: Change to `if (message.command !== "XML") console.log("Received message: ", message);`

File: src/activeLLM/suggestFix.js [hunk 4]
```diff
@@ -9,7 +9,19 @@ export async function suggestFix(
...
+    Now, here is a code snippet that violates this design rule: ${violation}
+    The violated code's file path is ${violationFilePath}
+    Can you suggest a fix to make this violation follow the given design rule?
+    Generate code with surrounding code included that follows the design rule.
+    Be sure to maintain proper whitespace with \\t and \\n.
+    Give a brief explanation of your fix as well.
+    Ensure that you include the fileName where the fix should be inserted at in the format Example.java.
+    Strictly output in JSON format. The JSON should have the following format:{"code": "...", "explanation": "...", "fileName": "..."}`;
+
+    /*const prompt = `Here is a design rule and its description: ${rule}
```
Comment (SaharMehrpour) [inline src/activeLLM/suggestFix.js:24]: We may forget why we have this commented prompt. Either add a comment describing it. Or even better remove it if we are not going to use it.

File: src/core/utilities.js (hunk 2 above)
Comment (SaharMehrpour) [reply src/core/utilities.js:101]: Edit this one too (for the readability of the log messages).

Comment (yaad96) [reply src/core/utilities.js]: done. please check

**Pull request merged**